p2p/switch.go

```diff
@@ -310,14 +308,11 @@ func (sw *Switch) reconnectToPeer(peer Peer) {
 		// sleep an exponentially increasing amount
 		sleepIntervalSeconds := math.Pow(reconnectBackOffBaseSeconds, float64(i))
 		sw.randomSleep(time.Duration(sleepIntervalSeconds) * time.Second)
-		peer, err := sw.DialPeerWithAddress(netAddr, true)
-		if err != nil {
-			sw.Logger.Info("Error reconnecting to peer. Trying again", "tries", i, "err", err, "peer", peer)
-			continue
-		} else {
-			sw.Logger.Info("Reconnected to peer", "peer", peer)
-			return
+		err := sw.DialPeerWithAddress(netAddr, true)
+		if err == nil {
+			return // success
```

Comment:
don't you think we should log success path too?

Reply:
yes it's logged within the DialPeerWithAddress function.

I wanted to avoid returning the peer because we weren't doing anything except logging it.

So I think it makes sense to let the function log it's success case, but the caller logs the error. Does that make sense or am I being silly ?

Reply:
maybe.. but I still think it's important to give user this sequence

```
[] .... Reconnecting to node A
[] .... Reconnected ...
```